SmartSaver.Domain/Repositories/Transaction/TransactionRepo.cs

```diff
@@ -0,0 +1,120 @@
+    {
+        public TransactionRepo(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        private readonly ApplicationDbContext _context;
+
+        public List<TransactionDto> GetThisMonthAccountSpendings(int accId)
+        {
+            var transactions = _context.Transactions
+                .Include(p => p.Category)
+                .Where(t => t.ActionTime > CurrentMonthFirstDayDate() &&
+                            t.AccountId == accId
+                            && t.Category.TypeOfIncome == false).ToList();
+            return transactions;
+        }
+
+        public List<Balance> GetBalanceHistory(int accId)
+        {
+            var chartData = new List<Balance>();
+
+            var transactions = _context.Transactions.Include(nameof(CategoryDto))
+                .Where(t => t.ActionTime > CurrentMonthFirstDayDate() && t.AccountId == accId);
+
+            decimal balance = 0;
+            foreach (var t in transactions)
+            {
+                balance += t.Amount;
+                chartData.Add(new Balance()
+                {
+                    Amount = balance,
+                    ActionTime = t.ActionTime
+                });
+            }
+
+            return chartData;
+        }
+
+        private DateTime CurrentMonthFirstDayDate()
+        {
+            var date = new DateTime();
+            var thisMonth = date.Month;
+            var thisYear = date.Year;
+
+            return new DateTime(thisYear, thisMonth, 1);
+        }
+
+        public List<TransactionDto> GetByAccountId(int accId)
+        {
+            return _context.Transactions
+                    .Include(p => p.Category)// Includes Category object.
+                    .Where(t => t.AccountId == accId)
+                    .OrderByDescending(a => a.ActionTime) // Order transactions from newest to oldest.
+                    .ToList();
+        }
+
+        public IEnumerable<TransactionDto> GetByAccountForDateRange(int accId, DateTime startData, DateTime endDate)
+        {
+            return _context.Transactions
+                    .Include(p => p.Category)// Includes Category object.
+                    .Where(t => t.AccountId == accId && t.ActionTime >= startData && t.ActionTime <= endDate)
+                    .OrderByDescending(a => a.ActionTime)
+                    .AsEnumerable(); // Order transactions from newest to oldest.
+        }
+
+        public async Task<int> CreateTransaction(TransactionDto transaction)
+        {
+            if (transaction.Amount == 0)
+            {
+                throw new InvalidModelException("Amount can not be 0");
+            }
+
+            transaction.Category = _context.Categories.First(c => c.Id == transaction.CategoryId);
+            transaction.ActionTime = DateTime.UtcNow;
+
+            if (!transaction.Category.TypeOfIncome)
+            {
+                transaction.Amount *= -1;
+            }
```

Comment:
With this if clause we can pass positive numbers only. It think the method is used in web api meaning that it has to accept negative numbers as well? + or - sign at the beginning of the number can determine what type of income it is